OutOfSchool/OutOfSchool.WebApi/Services/ProviderService.cs

```diff
@@ -178,6 +181,96 @@ public async Task<ProviderDto> Update(ProviderDto providerDto, string userId)
         public async Task<Guid> GetProviderIdForWorkshopById(Guid workshopId) =>
             await workshopServiceCombiner.GetWorkshopProviderId(workshopId).ConfigureAwait(false);
 
+        public async Task<ProviderStatusDto> UpdateStatus(ProviderStatusDto dto, string userId)
+        {
+            _ = dto ?? throw new ArgumentNullException(nameof(dto));
+
+            var provider = await providerRepository.GetById(dto.ProviderId).ConfigureAwait(false);
+
+            if (provider is null)
+            {
+                logger.LogError($"Provider(id) {dto.ProviderId} not found. User(id): {userId}");
+
+                return null;
+            }
+
+            provider.Status = dto.Status;
+            await providerRepository.UnitOfWork.CompleteAsync().ConfigureAwait(false);
+
+            logger.LogInformation($"Provider(id) {dto.ProviderId} Status was changed to {dto.Status}");
+
+            await SendNotification(provider, NotificationAction.Update, true, false).ConfigureAwait(false);
+
+            return dto;
+        }
+
+        public async Task<ProviderLicenseStatusDto> UpdateLicenseStatus(ProviderLicenseStatusDto dto, string userId)
+        {
+            _ = dto ?? throw new ArgumentNullException(nameof(dto));
+
+            var provider = await providerRepository.GetById(dto.ProviderId).ConfigureAwait(false);
+
+            if (provider is null)
+            {
+                logger.LogError($"Provider(id) {dto.ProviderId} not found. User(id): {userId}");
+
+                return null;
+            }
+
+            if (string.IsNullOrEmpty(provider.License) && dto.LicenseStatus != ProviderLicenseStatus.NotProvided)
+            {
+                throw new ArgumentException("Provider license is not provided. It cannot be approved.");
```

Comment:
here is a good place for log, though it's an expected part of the application flow, so info might be enough.